src/Pate/Memory/MemTrace.hs

```diff
@@ -1203,6 +1213,13 @@ readMemState :: forall sym ptrW ty.
   IO (RegValue sym (MS.ToCrucibleType ty))
 readMemState sym mem baseMem ptr repr = go 0 repr
   where
+  isPtrRead :: Bool
+  isPtrRead = case repr of
```

Comment:
Can we clarify that this is *potentially* a pointer read (because it is of pointer size) and not a guarantee that it is actually a pointer